target-chains/cosmwasm/tools/src/deployer/injective.ts

```diff
@@ -0,0 +1,177 @@
+import { readFileSync } from "fs";
+import { Bech32, toHex } from "@cosmjs/encoding";
+import { zeroPad } from "ethers/lib/utils.js";
+import assert from "assert";
+import { getNetworkInfo, Network } from "@injectivelabs/networks";
+import {
+  DEFAULT_STD_FEE,
+  MsgStoreCode,
+  MsgInstantiateContract,
+  PrivateKey,
+  TxGrpcClient,
+  TxResponse,
+  Msgs,
+  MsgMigrateContract,
+  createTransactionForAddressAndMsg,
+} from "@injectivelabs/sdk-ts";
+import { Deployer } from ".";
+
+export type InjectiveHost = {
+  network: Network;
+};
+
+export class InjectiveDeployer implements Deployer {
+  network: Network;
+  wallet: PrivateKey;
+
+  constructor(network: Network, wallet: PrivateKey) {
+    this.network = network;
+    this.wallet = wallet;
+  }
+
+  private injectiveAddress(): string {
+    return this.wallet.toBech32();
+  }
+
+  private async signAndBroadcastMsg(
+    msg: Msgs | MsgMigrateContract,
+    fee = DEFAULT_STD_FEE
+  ): Promise<TxResponse> {
+    const networkInfo = getNetworkInfo(this.network);
+
+    const { signBytes, txRaw } = await createTransactionForAddressAndMsg({
+      // @ts-ignore
+      message: msg,
+      address: this.injectiveAddress(),
+      endpoint: networkInfo.rest,
+      chainId: networkInfo.chainId,
+      fee,
+      pubKey: this.wallet.toPublicKey().toBase64(),
+    });
+
+    const sig = await this.wallet.sign(Buffer.from(signBytes));
+
+    /** Append Signatures */
+    txRaw.setSignaturesList([sig]);
+
+    const txService = new TxGrpcClient(networkInfo.grpc);
+    const txResponse = await txService.broadcast(txRaw);
+
+    if (txResponse.code !== 0) {
+      console.error(`Transaction failed: ${txResponse.rawLog}`);
+    } else {
+      console.log(
+        `Broadcasted transaction hash: ${JSON.stringify(txResponse.txHash)}`
+      );
+    }
+
+    return txResponse;
+  }
+
+  async deployArtifact(artifact: string): Promise<number> {
+    const contract_bytes = readFileSync(artifact);
+    console.log(`Storing WASM: ${artifact} (${contract_bytes.length} bytes)`);
+
+    const store_code = MsgStoreCode.fromJSON({
+      sender: this.injectiveAddress(),
+      wasmBytes: contract_bytes,
+    });
+
+    const txResponse = await this.signAndBroadcastMsg(store_code, {
+      amount: [
+        {
+          amount: String(500000000 * 5000000),
+          denom: "inj",
+        },
+      ],
+      gas: "5000000",
+    });
+
+    var codeId: number;
+    try {
+      const ci = /"code_id","value":"\\"([^\\"]+)/gm.exec(txResponse.rawLog)![1];
```

Comment:
maybe extract a utility function to pull out fields from the txResponse log? I think you could do it generically for the code_id here and the contract_address below, plus some similar logic occurs in other places in the PR.